tests/evaluate/test_default_evaluator.py

```diff
@@ -3582,3 +3582,44 @@ def test_target_prediction_col_mapping():
                 "correctness/v1/variance": 0.0,
                 "correctness/v1/p90": 3.0,
             }
+
+
+def test_evaluate_custom_metric_with_string_type():
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            artifact_path="model", python_model=language_model, input_example=["a", "b"]
+        )
+        data = pd.DataFrame({"text": ["Hello world", "My name is MLflow"]})
+        from mlflow.metrics import make_metric
+
+        def word_count_eval(predictions):
+            scores = []
+            avg = 0
+            aggregate_results = {}
+            for prediction in predictions:
+                scores.append(prediction)
+                avg += len(prediction.split(" "))
+
+            avg /= len(predictions)
+            aggregate_results["avg_len"] = avg
+
+            return MetricValue(
+                scores=scores,
+                aggregate_results=aggregate_results,
+            )
```

Comment:
test looks fine, but i am not sure why a user would construct such an eval_fn like this.